http_client.go

```diff
@@ -18,20 +18,57 @@
 
 package promconfig
 
+import "net/url"
+
 // HTTPClientConfig configures an HTTP client.
 type HTTPClientConfig struct {
 	// The HTTP basic authentication credentials for the targets.
-	BasicAuth *BasicAuth `yaml:"basic_auth,omitempty"`
-	// The bearer token for the targets.
-	BearerToken string `yaml:"bearer_token,omitempty"`
-	// The bearer token file for the targets.
-	BearerTokenFile string `yaml:"bearer_token_file,omitempty"`
+	BasicAuth *BasicAuth `yaml:"basic_auth,omitempty" json:"basic_auth,omitempty"`
```

Comment:
did you add `json` here for a reason?
It adds changes in our test which is unexpected.

Reply:
Also `json` should be added either everywhere or nowhere.

Reply:
The `json` just came as part of the copy-paste I did from Prometheus structs. I will remove the tags as I didn't realize this project stripped these. Will also fix the tests.